lib/puppet/pops/loader/module_loaders.rb

```diff
@@ -227,6 +235,10 @@ def to_s()
     end
 
     def add_to_index(smart_path)
+      # TODO:  testing to see if this is necessary to wrap smart_path.generic_path with Puppet::FileSystem.expand_path
+      not_expanded = (smart_path.generic_path.index('~'))
+      stack = not_expanded ? caller.find_all { |call| call =~ /^C\:\/source\/puppet\/(lib|spec)\// }.join("\n") : ''
+      puts("\nAbstractPathBasedModuleLoader#add_to_index - Unexpanded path #{smart_path.generic_path} - call stack:\n#{stack}\n") if not_expanded
```

Comment:
The above three lines seems to occur at least 9 times in this PR and the only difference seems to the text and path and some difference in how `not_expanded` is computed. Perhaps consider turning that into a utility method (differences in how not_expanded is computed could be provided as a default implementation and an optional block).